geolocation/src/definitions.ts

```diff
@@ -0,0 +1,139 @@
+declare module '@capacitor/core' {
+  interface PluginRegistry {
+    Geolocation: GeolocationPlugin;
+  }
+}
+
+export type CallbackID = string;
+
+export type GeolocationPermissionType = 'location';
+export type GeolocationState = PermissionState;
+
+export interface GeolocationPermissionStatus {
+  location: GeolocationState;
+}
+
+export interface GeolocationPlugin {
+  /**
+   * Get the current GPS location of the device
+   *
+   * @since 1.0.0
+   */
+  getCurrentPosition(
+    options?: GeolocationOptions,
+  ): Promise<GeolocationPosition>;
+
+  /**
+   * Set up a watch for location changes. Note that watching for location changes
+   * can consume a large amount of energy. Be smart about listening only when you need to.
+   *
+   * @since 1.0.0
+   */
+  watchPosition(
+    options: GeolocationOptions,
+    callback: GeolocationWatchCallback,
+  ): CallbackID;
+
+  /**
+   * Clear a given watch
+   */
+  clearWatch(options: { id: string }): Promise<void>;
+
+  /**
+   * Check location permissions
+   *
+   * @since 1.0.0
+   */
+  checkPermissions(): Promise<GeolocationPermissionStatus>;
+
+  /**
+   * Request location permissions
+   *
+   * @since 1.0.0
+   */
+  requestPermissions(options: {
+    permissions: GeolocationPermissionType[];
+  }): Promise<GeolocationPermissionStatus>;
```

Comment:
The only way of calling the function is like this `Geolocation.requestPermissions({permissions: ['location']})`, since permissions allows an array of `GeolocationPermissionType`, but `GeolocationPermissionType` only allows `'location'` value, so there is no point on using an array. And also, since only one alias can be checked, I don't think it's worth using options, just call it without parameters.
```suggestion
  requestPermissions(): Promise<GeolocationPermissionStatus>;
```